databuilder/file_formats/arrow.py

```diff
@@ -65,7 +65,11 @@ def get_field_and_convertor(name, spec):
         type_ = PYARROW_TYPE_MAP[spec.type]()
 
     if spec.categories is not None:
-        index_type = smallest_int_type_for_range(0, len(spec.categories) - 1)
+        # Although dictionary indices can obviously never be negative we use `-1` as the
```

Comment:
I might have misunderstood. I thought `dict` indices could be any hashable type, which includes negative integers:

```python
>>> d = {-1: "minus one"}
>>> d[-1]
'minus one'
```

Reply:
Ah yeah, two different sorts of dictionary here: Python ones and PyArrow ones. The PyArrow dictionary index is just the offset into the dictionary array and so is necessarily positive. I can make the comment clearer.